bpf/sockops/bpf_sockops.c

```diff
@@ -29,6 +29,10 @@ static __always_inline void sk_extract4_key(const struct bpf_sock_ops *ops,
 	key->dip4 = ops->remote_ip4;
 	key->sip4 = ops->local_ip4;
 	key->family = ENDPOINT_KEY_IPV4;
+	/* We will get misses on UDP, but the verifier
```

Comment:
Shall we document this limitation? Also, have you considered having two cases:
```
#ifdef CAN_ACCCESS_SK_FIELDS_FOO
key->protocol = ...;
#else
key->protocol = IPPROTO_TCP;
#endif
```

Reply:
Yeah. That could work. Would you mind if I created an issue for this? I would like to really nail down **exactly** which versions of the kernel the verifier allowed access to the socket object for these programs. I assume we have a kernel version check somewhere in the go code?

Reply:
Sure, just let's create an issue with follow-up items first.

Reply:
@brb were you looking to also insert the GH issue number into the code comments so we can refer back in future? Or you're fine with just filing the issue & following up?

Reply:
I'm fine with just creating the issue. I want to avoid a case when we forget to create an issue and to be addressed comments get forgotten.

Reply:
Here's the issue:
https://github.com/cilium/cilium/issues/13490